### DISCUSSION TITLE
Efficient dashboard

### DISCUSSION THREAD

**Discussion by aldbr** [DIRACGrid/diracx-web#63] [User personas and stories]
### Link to User Story

https://github.com/DIRACGrid/diracx-web/discussions/62

### Proposed Design Solution

Users should be able to open multiple instances of each application, for instance `Job Monitor (1)` and `Job Monitor (2)`. They should be able to rename the instances to easily identify them.

They should be able to group them according to specific criteria. They should be able to customize their dashboard/groups/instances according to their needs. Dashboard/groups/instances should be shareable (via a URL?)

Depending on their privileges, users should see all or a subset of the available applications. For instance, a normal user would probably only have access to the `Job Monitor` application.

Basically, a normal user - which would represent the vast majority of the users - would only need 1/2 instances, whereas a DIRAC administrator would need all applications and would open many different instances of each of them (and would group them).

2 main solutions are proposed here:
-  The DIRAC Webapp approach: a side bar with all the applications available to the connected user (a normal user would only see 1/2 applications). When the user would want to open an instance of application, he/she would click on it in the side bar, which would open a new horizontal tab. The user would be able to save the opened tabs to create a "Desktop" that would then appear below the list of applications.
   - Pros: 
     - Users can open applications directly from the sidebar, which is straightforward and quick.
   - Cons: 
     - With the addition of multiple tabs, the interface may become cluttered, especially for admins who work with many instances.
     - Such design is not adapted to mobile device. We would need to adapt it, probably by creating a right side bar on mobile screens?

![Screenshot from 2023-12-06 14-28-46](https://github.com/DIRACGrid/diracx-web/assets/19813740/d51d3e3b-9e2d-4f3f-b810-c026d9dea1ae)

-  A novel approach: a side bar which contains the instances of the applications. When a user would want to add an instance to the side bar, he/she would have to click on the button "Add an application", which would open a new window, where he/she would select the application, which would be added to the side bar. The user could then organize the opened instances into different groups directly from the side bar.
   - Pros: 
     - The ability to organize instances into groups directly from the sidebar could provide a more dynamic way of managing tasks.
     - There would be no huge UX difference from a mobile device. 
   - Cons: 
     - Having to add applications through an additional window might slow down the workflow.

![Screenshot from 2023-12-06 14-29-17](https://github.com/DIRACGrid/diracx-web/assets/19813740/05af4d5f-7712-46ca-8cbe-43ba2a75b626)

We could also imagine some variants:
- a search bar at the top to quickly find a specific instance based on its name
- applications could be directly accessible from the top bar in the "novel approach": this would be the exact opposite of the "DIRAC Webapp approach".

Comments are more than welcome here!

**Vote for your preference**

To vote, react to the post with the following emojis:
- :tada: for the first approach
- :rocket: for the second approach

Double vote does not count.

**Comment by marianne013** [comment]
I have no fixed opinion -- does that count ? --Daniela
